internal/services/mongocluster/mongo_cluster_firewall_rule_resource_test.go

```diff
@@ -0,0 +1,157 @@
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+type MongoClusterFirewallRuleResource struct{}
+
+func TestAccMongoClusterFirewallRule_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_mongo_cluster_firewall_rule", "test")
+	r := MongoClusterFirewallRuleResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccMongoClusterFirewallRule_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_mongo_cluster_firewall_rule", "test")
+	r := MongoClusterFirewallRuleResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func TestAccMongoClusterFirewallRule_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_mongo_cluster_firewall_rule", "test")
+	r := MongoClusterFirewallRuleResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func (r MongoClusterFirewallRuleResource) Exists(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := firewallrules.ParseFirewallRuleID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := clients.MongoCluster.FirewallRulesClient.Get(ctx, *id)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving %s: %+v", *id, err)
+	}
+
+	return pointer.To(resp.Model != nil), nil
+}
+
+func (r MongoClusterFirewallRuleResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_mongo_cluster_firewall_rule" "test" {
+  name             = "acctest-mcfr-%d"
+  mongo_cluster_id = azurerm_mongo_cluster.test.id
+  start_ip_address = "10.0.0.1"
+  end_ip_address   = "10.0.0.255"
+}
+`, r.template(data), data.RandomInteger)
+}
+
+func (r MongoClusterFirewallRuleResource) requiresImport(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_mongo_cluster_firewall_rule" "import" {
+  name             = azurerm_mongo_cluster_firewall_rule.test.name
+  mongo_cluster_id = azurerm_mongo_cluster_firewall_rule.test.mongo_cluster_id
+  start_ip_address = azurerm_mongo_cluster_firewall_rule.test.start_ip_address
+  end_ip_address   = azurerm_mongo_cluster_firewall_rule.test.end_ip_address
+}
+`, r.basic(data))
+}
+
+func (r MongoClusterFirewallRuleResource) complete(data acceptance.TestData) string {
```

Comment:
Same as above, this should prolly be called `update` instead of `complete`...
```suggestion
func (r MongoClusterFirewallRuleResource) update(data acceptance.TestData) string {
```